clippy_lints/src/write.rs

```diff
@@ -234,20 +235,22 @@ fn check_tts<'a>(cx: &EarlyContext<'a>, tts: &ThinTokenStream, is_write: bool) -
         false,
         false,
     );
-    if is_write {
```

Comment:
You cannot remove `is_write`. Maybe a more in depth explanation, what is happening here might help you:

`check_tts` is called on both macros `print(ln)!()` and `write(ln)!()`. The arguments of `writeln!(arg1[, arg2[, fmt_args]])` are 
1. `arg1` the write buffer
2. `arg2` (optional) the string to write
3. `fmt_args` (optional) not important here

`println!()` only has the last 2 arguments.

This means calling `parse_expr()` on `println!("something")` would result in the string to print and could result in some weird behavior:

`parse_expr()` returns a `Result<Expr, _>`. If this fails, we return from the function, see `.ok()?`. Since `println!()` is valid (but `writeln!()` isn't), we would skip every `println!()` without arguments here.

`expect(&token::Comma)` checks if `writeln!()` has a second argument. If it has no second argument (`expect()` errors) we are good and can return `None` (`.ok()?`) and don't lint.

This behavior should not change!

What to do here:
match `parser.parse_expr().map_err(|mut err| err.cancel()).ok()` to `Some(expr)` and set the `expr` variable to `Some(expr)`, else return `(None, _)`.

Everything else does not need to change here.